src/Cassandra/Metadata.cs

```diff
@@ -133,14 +148,118 @@ public IEnumerable<IPEndPoint> AllReplicas()
             return Hosts.AllEndPointsToCollection();
         }
 
-        internal void RebuildTokenMap()
+        internal async Task<bool> RebuildTokenMapAsync(bool retry)
+        {
+            var currentTicks = DateTime.UtcNow.Ticks;
```

Comment:
`Stopwatch` is not threadsafe though, I would have to wrap the call with a `lock`, which doesn't seem like a great idea.

Reply:
I've edited the message to make it more clear, I've meant to avoid using system clock to measure time and use high resolution counters instead (when available). Using `DateTime.UtcNow` might get you a resolution of ~16 milliseconds (1000 / 60), which is not very good.